src/Authoring/WinRT.SourceGenerator/DiagnosticUtils.cs

```diff
@@ -0,0 +1,619 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Generic;
+using System.Linq;
+using WinRT.SourceGenerator;
+
+namespace Generator 
+{
+    public class WinRTRules
+    {
+
+        #region ModifierHelpers
+        private bool SyntaxTokenIs(SyntaxToken stx, string str) { return stx.Value.Equals(str); }
+        
+        private bool ModifiersContains(SyntaxTokenList modifiers, string str)
+        {
+            foreach (var modifier in modifiers)

```

Comment:
Technically this is similar to modifiers.Any(modifier => modifier.ValueText == str);

Reply:
I'll try this out, thanks!